lib/dirac_coarse.cpp

```diff
@@ -138,16 +150,26 @@ namespace quda {
     int pad = std::max( { (x[0]*x[1]*x[2])/2, (x[1]*x[2]*x[3])/2, (x[0]*x[2]*x[3])/2, (x[0]*x[1]*x[3])/2 } );
     gParam.pad = gpu ? gParam.nFace * pad * 2 : 0; // factor of 2 since we have to store bi-directional ghost zone
 
-    if (gpu) Y_d = new cudaGaugeField(gParam);
-    else     Y_h = new cpuGaugeField(gParam);
+    if (gpu) {
+      Y_d = std::make_shared<cudaGaugeField>(gParam);
+      GaugeFieldParam milcParam(*Y_d);
+      milcParam.order = QUDA_MILC_GAUGE_ORDER;
+      Y_aos_d = std::make_shared<cudaGaugeField>(milcParam);
```

Comment:
I'm worried about the memory overheads of _always_ creating this structure in cases where the MMA routines don't end up getting used (i.e., I may try to avoid when I'm testing naive staggered KD), is there a sane way to toggle this?

Reply:
Now the AoS fields won't be there if no MMA is used. :)